gnuradio-runtime/include/gnuradio/hier_block2.h

```diff
@@ -45,6 +45,7 @@ class GR_RUNTIME_API hier_block2 : public basic_block
      */
     std::unique_ptr<hier_block2_detail> d_detail;
 
+    std::vector<int> d_affinity; // thread affinity proc. mask
```

Comment:
why keep this in the interface instead of in the detail, if it's private anyways?

Reply:
I was thinking the same thing ... but there are worse offenders in this file.

Reply:
It's the same behaviour as in block.k. See line 728 and 799

Reply:
And you need it to save and clear the mask if you call the affinity functions for the hier block.